app/controllers/admin/orders_controller.rb

```diff
@@ -77,6 +77,6 @@ def order_params
   end
 
   def search_params
-    params.fetch(:search_term, {}).permit(:order_id, :ship_email, :ship_phone)
+    params.fetch(:order_search_term, {order_id: nil, ship_email: nil, ship_phone: nil}).permit(:order_id, :ship_email, :ship_phone)
```

Comment:
when url is just http://localhost:3000/admin/orders/search, fetch will return the replaced hash when :order_search_term not exist

and` {order_id: nil, ship_email: nil, ship_phone: nil} `is for Placeholder condition 

`where('ship_phone = :ship_phone OR ship_email = :ship_email OR orders.id = :order_id', search_term)`

this require all the keys exist in search_term hash, if just pass {} , it will fail

http://guides.rubyonrails.org/action_controller_overview.html#more-examples
http://stackoverflow.com/questions/25914514/rails-4-strong-parameters-without-required-parameters